### PR TITLE
Add vkcs_networking_sdn data source

### PR DESCRIPTION
**Pull request by morozovalekseywot**
Add vkcs_networking_sdn data source for getting available SDN for the current project

### CODE REVIEW

File: vkcs/networking/data_source_sdn.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+func DataSourceNetworkingSDN() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceNetworkingSDNRead,
+
+		Schema: map[string]*schema.Schema{
+			"sdn": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Names of available VKCS SDN's in the current project.",
```
Comment (paaanic) [inline vkcs/networking/data_source_sdn.go:21]:
```suggestion
				Description: "Names of available VKCS SDNs in the current project.",
```

File: vkcs/networking/data_source_sdn.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	if err != nil {
+		return diag.Errorf("Error creating VKCS networking client: %s", err)
+	}
+
+	sdn, err := isdn.GetAvailableSDNs(networkingClient)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(config.GetTenantID())
```
Comment (paaanic) [inline vkcs/networking/data_source_sdn.go:40]: Common way to do it is to set it to some random string, for example, string representation of unix time.

File: vkcs/networking/data_source_sdn.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"sdn": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Names of available VKCS SDN's in the current project.",
+			},
+		},
+		Description: "Use this data source to get list of an available VKCS SDN in the current project.",
```
Comment (paaanic) [inline vkcs/networking/data_source_sdn.go:24]:
```suggestion
		Description: "Use this data source to get a list of available VKCS SDNs in the current project.",
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,9 @@ description: |-
 
 # VKCS Provider's changelog
 
+#### v0.7.4 (unreleased)
+- Add vkcs_networking_sdn data source for getting available SDN's.
```
Comment (paaanic) [inline CHANGELOG.md:11]:
```suggestion
- Add vkcs_networking_sdn data source for getting a list of available SDNs.
```

**Review by paaanic — CHANGES_REQUESTED** [review]

File: vkcs/networking/data_source_sdn.go (hunk 2 above)
Comment (morozovalekseywot) [reply vkcs/networking/data_source_sdn.go:40]: I discussed this with @ftersin , we decided that it was better to set the project id

File: vkcs/internal/services/networking/sdn.go [hunk 5]
```diff
@@ -21,3 +27,24 @@ func SelectSDN(c *gophercloud.ServiceClient, sdn string) error {
 	}
 	return nil
 }
+
+func sdnURL(c *gophercloud.ServiceClient) string {
```
[the remark this answers is not in the document]
Comment (morozovalekseywot) [reply vkcs/internal/services/networking/sdn.go]: done

File: vkcs/networking/data_source_sdn.go (hunk 1 above)
Comment (morozovalekseywot) [reply vkcs/networking/data_source_sdn.go]: done

File: vkcs/networking/data_source_sdn.go (hunk 2 above)
Comment (ftersin) [reply vkcs/networking/data_source_sdn.go:40]: it does not matter in my opinion

**Review by ftersin — CHANGES_REQUESTED** [review]
add a doc template and an examle

File: vkcs/networking/data_source_sdn.go (hunk 2 above)
Comment (paaanic) [reply vkcs/networking/data_source_sdn.go:40]: I simply think that it isn't a good place to get tied to some authentication method, for example, there may be a case when we are authenticating through a token with no authentication scope set, and then the tenant ID will be empty.

Comment (morozovalekseywot) [reply vkcs/networking/data_source_sdn.go:40]: maybe then `config.GetTenantID() + hashcode.Strings(sdn)`?

Comment (paaanic) [reply vkcs/networking/data_source_sdn.go]: I don't see any reason to use tenant ID as id here, thus, it's just easier to stick to the common way:
https://github.com/vk-cs/terraform-provider-vkcs/blob/727f9b7c808891302bba7c75cf3d20645f19d550/vkcs/regions/data_source_regions.go#L58

File: examples/networking/sdn/main.tf [hunk 6]
```diff
@@ -0,0 +1,2 @@
+data "vkcs_networking_sdn" "sdn" {
+}
```
Comment (ftersin) [inline examples/networking/sdn/main.tf:2]: add an empty line

File: vkcs/networking/data_source_sdn.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"sdn": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Names of available VKCS SDNs in the current project.",
+			},
+		},
+		Description: "Use this data source to get a list of available VKCS SDNs in the current project.",
```
Comment (ftersin) [inline vkcs/networking/data_source_sdn.go:25]: lets add
The first SDN is default. You do not have to specify default sdn argument in resources and datasources. You may specify non default SDN only for root resources such as `vkcs_networking_router`, `vkcs_networking_network`, `vkcs_networking_secgroup` (they do not depend on any other resource/datasource with sdn argument).

Comment (morozovalekseywot) [reply vkcs/networking/data_source_sdn.go]: add

File: examples/networking/sdn/main.tf (hunk 6 above)
Comment (morozovalekseywot) [reply examples/networking/sdn/main.tf]: done

File: CHANGELOG.md (hunk 4 above)
Comment (morozovalekseywot) [reply CHANGELOG.md]: done

File: vkcs/networking/data_source_sdn.go (hunk 2 above)
Comment (morozovalekseywot) [reply vkcs/networking/data_source_sdn.go]: done

**Review by ftersin — APPROVED** [review]

**Review by paaanic — APPROVED** [review]

**Pull request merged**